packages/languages/src/browser/language-client-contribution.ts

```diff
@@ -38,7 +38,8 @@ export abstract class BaseLanguageClientContribution implements LanguageClientCo
     constructor(
         @inject(Workspace) protected readonly workspace: Workspace,
         @inject(Languages) protected readonly languages: Languages,
-        @inject(LanguageClientFactory) protected readonly languageClientFactory: LanguageClientFactory
+        @inject(LanguageClientFactory) protected readonly languageClientFactory: LanguageClientFactory,
+        @inject(Commands) protected readonly commands: Commands
```

Comment:
you can use the property injection to avoid changing constructors of subclasses

Reply:
Aren't we using constructor injection by default all over the place?

Reply:
I don't think we have such rule

Reply:
We at least used to have it and somehow gave it up last month without explicit discussion :)
But I like field injection, too, sometimes.